pkg/platform/runtime2/runtime.go

```diff
@@ -42,17 +46,23 @@ func New(target setup.Targeter) (*Runtime, error) {
 		return rt, &NeedsUpdateError{errs.New("Runtime requires setup.")}
 	}
 
-	analytics.Event(setup.CatRuntime, setup.ActCache)
+	if noSetupYet {
+		analytics.Event(analytics.CatRuntime, analytics.ActRuntimeCache)
+	}
```

Comment:
Similar issue as above.  But this is even more tricky, as it is not only about sending an event only once.  When called from the `Update()` function, we don't want to send this event ever, even if it was the first time, we tried...

Reply:
Actually, what do you think about adding a `runtime.NewWithUpdate()` function that does all the things: trying to get a runtime from cache, and if not available it calls the `Update()` method (which then could become private).  Those two functions would be more straight-forward to instrument without this awkward `noSetupYet` flag.  It would also reduce code duplication, as we essentially have the `NewWithUpdate()` function written explicitly in several runners right now...